### DISCUSSION TITLE
Merged epub from multiple epubs show 1st page as blank?

### DISCUSSION THREAD

**Discussion by mozbugbox** [pymupdf/PyMuPDF#791] [Looking for help] [answered]
I have an epub created from multiple epub files using the `epubmerger` plugin of `Calibre`.

When load the epub with `PyMuPDF` v1.17, the first page was displayed as a blank image. Only the 2nd page displayed the cover image properly.

While test with `mupdf` viewer, the first page showed the cover image. Dump the page 0 with `mutool convert` also showed the cover image properly. 

Is there any work around to shows the 1st page as what mupdf does? 

A test epub was attached.

[test-merge.zip](https://github.com/pymupdf/PyMuPDF/files/5742651/test-merge.zip)

**Comment by JorjMcKie** [comment, marked as answer]
Confirming this behaviour - weird indeed.
Page rect is (0, 0, 450, 600) when opened without parameter. But if you do this:
```python
>>> import fitz
>>> doc=fitz.open("test-merge.epub", rect=(0,0,400,600))
>>> page=doc[0]
>>> page.rect
Rect(0.0, 0.0, 400.0, 600.0)
>>> pix=page.get_pixmap()
>>> pix.writeImage("t.png")
>>> 
```
the page picture is shown correctly. It also works if choosing an A4 rectangle on open.
One can also do a `doc.layout()` with an appropriate rectangle before loading the first page, and it will work ...

**Reply by JorjMcKie** [reply]
In the next version, I will do an automatic layouting with `fitz.Rect(0, 0, 400, 600), fontsize=11` if the document is reflowable and no layout args are given on open.

**Reply by JorjMcKie** [reply]
It seems like any rectangle works that has a different aspect ratio than the front image ...

**Comment by mozbugbox** [comment]
Thanks for your investigation. You are right, the aspect ratio will change the behavior.

It seems that  re-layout() the doc after loading the first page works well.

However, what aspect ratio that will make the cover page work is not clear to me. A rect (500, 600) don't work. (450, 550) don't either. A more portrait aspect works, while landscape don't. Don't know why.

**Reply by JorjMcKie** [reply]
Well, maybe my assumption was a bit too bold.
If layouting with 400 x 600 page size it works for sure.
